src/main/java/Snake/GamePanel.java

```diff
@@ -156,9 +156,18 @@ public void move() {
         // TODO: You will only need to edit between these comments
 
         // Edit here
-
-
-        // TODO: You will only need to edit between these comments
+        int prevX = x[0];
+        int prevY = y[0];
+        int tmpX;
+        int tmpY;
+        for (int i = 1; i < bodyParts; i++) {
+            tmpX = x[i];
+            tmpY = y[i];
+            x[i] = prevX;
+            y[i] = prevY;
+            prevX = tmpX;
+            prevY = tmpY;
```

Comment:
This can be cleaned up a bit using index - 1. You don't have to use so many variables. 

y[i] = y[i-1]; 
x[i] = x[i-1]; 

This might help.